storage/range_tree.go

```diff
@@ -83,151 +90,178 @@ func getRangeTree(db *client.KV) (*rangeTree, error) {
 	if !ok {
 		return nil, util.Errorf("Could not find the range tree:%s", engine.KeyRangeTreeRoot)
 	}
-	return (*rangeTree)(tree), nil
+
+	return &treeContext{
+		db:          db,
+		tree:        tree,
+		treeUpdated: false,
+		nodes:       map[string]cachedNode{},
+	}, nil
+}
+
+// setRoot sets the tree root key in the cache. It also marks the root for
+// writing during a flush.
+func (t *treeContext) setRootKey(key *proto.Key) {
+	t.tree.RootKey = *key
+	t.treeUpdated = true
 }
 
-// getRangeTreeNode return the RangeTree node for the given key.  If the key is
-// nil, an empty RangeTreeNode is returned.
-func getRangeTreeNode(db *client.KV, key *proto.Key) (*rangeTreeNode, error) {
+// setNode sets the node in the cache so all subsequent reads will read this
+// upated value.  It also marks the node for writing during a flush.
```

Comment:
Yeah.  I was taught 2 spaces after a period (or colon) in typing class.  I would still argue that's correct for any monospaced font.  But I'll try to curb my habit.